File: backend/src/main/kotlin/bo/edu/umsa/backend/dto/ProjectReportDto.kt
package bo.edu.umsa.backend.dto

import java.sql.Timestamp

data class ProjectReportDto(
    val projectId: Int,
    val projectName: String,
    val projectStatusName: String,
    val projectFinishedTasks: Int,
    val projectTotalTasks: Int,
    val projectDateFrom: Timestamp,
    val projectDateTo: Timestamp,
    val projectEndDate: Timestamp?,
    val projectOwners: List<String>,
    val projectModerators: List<String>,
    val projectMembers: List<String>,
    val taskReports: List<TaskReportDto>,
)

File: frontend/src/app/features/report/models/project-report.dto.ts
import {TaskReportDto} from "./task-report.dto";

export interface ProjectReportDto {
    projectId: number;
    projectName: string;
    projectStatusName: string;
    projectFinishedTasks: number;
    projectTotalTasks: number;
    projectDateFrom: Date;
    projectDateTo: Date;
    projectEndDate: Date;
    projectOwners: string[];
    projectModerators: string[];
    projectMembers: string[];
    taskReports: TaskReportDto[];
}
